src/components/Calculator.js

```diff
@@ -12,29 +12,51 @@ export const Calculator = () => {
   const numbers = '0123456789';
 
   const handleValues = (event) => {
-    if (numbers.includes(event.target.name)) {
+    /* validate if we got a operator as first value. set that value to the first
+    other set the operator */
+
+    if (operators.includes(event.target.name)) {
       if (firstValue === '') {
         setFirstValue(event.target.name);
-        setResult(`${result}${event.target.name}`);
       } else {
-        setSecondValue(event.target.name);
-        setResult(`${result}${event.target.name}`);
+        setOperator(event.target.name);
       }
+      setResult(`${result}${event.target.name}`);
     }
 
-    if (operators.includes(event.target.name)) {
-      setOperator(event.target.name);
+    if (numbers.includes(event.target.name)) {
+      // validate for emp
+      if (dataObject.operation === '') {
+        // set the firstValue if we have non else concatenate the existing with new value
+        if (firstValue === '') {
+          setFirstValue(event.target.name);
+        } else if (firstValue) {
+          setFirstValue(`${firstValue}${event.target.name}`);
+        }
+      } else {
+        setSecondValue(event.target.name);
+      }
```

Comment:
- [ ] It would be better if we fix the display functionality of the `.` and `%` operations. Now the signs do not only show on the display, but it also does not work logically. It will be great if this is fixed

Reply:
@mmsesay, It was mentioned in the previous review as required so as to keep up with code review consistency